File: shared/R/enum.R
library(here)
library(jsonlite)

get_enum <- function(enum) {
  enums <- get_enums()
  if (!enum %in% names(enums)) stop("Invalid enum does not exist in enum.json.")
  enums[[enum]]
}

get_enums <- function() {
  enum_path <- here::here("shared", "enum.json")
  fromJSON(enum_path, simplifyVector = TRUE)
}

File: shared/python/enum.py
# package imports
import json

# function imports
from pathlib import Path

ENUM_PATH = Path(__file__).parent.parent / "enum.json"

def get_enum(enum: str) -> dict[str, str]:
    enums = get_enums()
    
    if enum not in enums:
        raise ValueError(f"{enum=} does not exist in enum.json.")
    
    return enums.get(enum, {})

def get_enums() -> dict[str, dict[str, str]]:
    with open(ENUM_PATH, "r", encoding="utf-8") as file:
        return json.load(file)